plugins/inputs/prometheus/prometheus.go

```diff
@@ -202,7 +202,7 @@ func (p *Prometheus) GetAllURLs() (map[string]URLAndAddress, error) {
 			return allURLs, err
 		}
 
-		for _, url := range getMesosTaskPrometheusURLs(tasks, p.mesosHostname) {
+		for _, url := range getMesosTaskPrometheusURLs(tasks) {
```

Comment:
Would it make sense to call `getTaskIP` above this line and pass in the IP, keeping this and the other function signatures the same? It would also only run the `getTaskIP` logic once as opposed to in multiple places.

I also did a cursory search of `p.mesosHostname` and it doesn't look like it's used anywhere anymore so maybe we can do a clean up and remove that field and [`getMesosHostname`](https://github.com/dcos/telegraf/pull/58/files#diff-7126ee4cf2b69635e3c56ae1cdb36e20R384)

Reply:
The change this PR makes is that instead of using the hostname to form the metrics source URL, we now use the task's ip address. So we no longer need to pass the hostname in. 

Each task can potentially have a different IP address. Since this method retrieves URLs from multiple tasks, passing it a single IP address won't work - it needs to use the IP address from each task. 

WRT your second point, you're quite right - I had thought it was used in `getMesosClient`, but it isn't. I'll remove it.

Reply:
Ahh, I see, missed that loop the first time round :). It looks like both `getEndpointsFromTaskPorts` and `getEndpointFromTaskLabels` get passed the same task, so maybe `getTaskIP` can just be called once on the task and passed into both of those functions to clean that up a bit and avoid the extra call - but it's not a big deal and I won't block on it 😸